### DISCUSSION TITLE
앱 테마 설정을 위한 UI 선정

### DISCUSSION THREAD

**Discussion by JMM00** [DeveloperAcademy-POSTECH/MC2-Team7-Larasy#105] [Ideas]
### 아이디어
- 날씨를 테마로 색상 테마 제공
- 사계절을 색으로 표현하여 제공
### 고려사항
- 테마를 고르기 위해 눌러야할 버튼을 기존 디자인에 맞춰서 에셋을 제작하여 추가할 지 설정 버튼을 넣을지.. -> 개인적으로는 분위기에 맞춰 시계 에셋을 제작해서 넣고 싶은데 손이 안따라주네요ㅠㅠ
- 다만 이 경우 각 색에 맞는 다크모드 색상의 에셋을 뽑기 어려워 다크모드는 하기 어려울 것 같습니다.
### 예시 이미지
<img src="https://user-images.githubusercontent.com/41153398/225951375-32f5c081-c3cf-4dab-8330-9930e8213d1f.png" width=300>

### 아이디어
- 기존에 존재하는 버섯 조명을 통한 다크모드 제공
### 고려사항
- 모든 에셋을 다크모드 색상으로 조정해서 뽑아야 합니다.
- 다양한 색상 테마 제공이 어려워집니다. (모든 색과 어울리는 다크모드 색상 찾기 어려움..)
### 예시 이미지
- 색상은 예시로 대충..뽑아서 전체적인 분위기만 봐주시길 바랍니다!
- 아래의 영상은 버섯 조명을 클릭하는 영상입니다.

https://user-images.githubusercontent.com/41153398/225953599-8edacc19-2e6a-4e27-8aa5-afa0ea6f16dc.mp4

**Comment by YunSeok-Choi** [comment]
와 저는 둘다 너무 이쁜데요???
1번, 2번 둘다 가져가고 싶네요ㅜㅜ

1번을 기본적으로 제공하면서, 2번은 이스터 에그 느낌으로 가져가면 어떨까요??
엘리가 고려하셨던 부분처럼 2번 다크모드가 사용자 글 작성 뷰에서도 적용된다면 1번과 같은 테마를 제공하지 못하니,
글 작성 뷰(Write View)에서만 다크모드가 아닌 기본 테마를 적용하고, 따로 버튼을 두어서 원하는 테마를 적용시키는 방식으로 말이죠!

사용자가 글을 작성할 때  작성 당시의 자신의 감정에 맞는 테마를 적용하고 싶어할 것같아요! 
따라서 옵션을 만들어서 한 테마를 적용하면 작성했던 홈 뷰를 포함한 작성한 모든 글들이 선택한 테마가 되는 것이 아니라, 
사용자가 글 작성시 선택한 테마에 맞춰서 저장되는 형식이죠!
 그럼 다크모드도 유지되면서 내가 작성한 글을 들어가면 선택한 테마가 적용되는 느낌으로..? (그러면 디자인이 조금 이상해질 수도 있겠네요ㅜ)
제가 적으면서도 공수가 많이 들 것같긴하네요...ㅎㅎ
만약 엘리가 생각한 방향이 아니시라면 피드백 주세요!

디자인이든 개발이든 도움이 필요하시면 말씀해주세요! 언제든지 달려가겠습니다🔥🔥

**Reply by JMM00** [reply]
둘다 어울리는 색으로...색을 한번 뽑아볼까요..? 괜찮다 생각되는 색상은 피그마에서 라라시>NewDesign>앱테마 만들어보기 에서 만들어주시면 반영하겠습니단!!

- 저희 이번 스프린트 원래 이슈가 이미지 저장 시의 테마였더군요...ㅠㅠ 그래서 이미지 저장 시의 테마를 우선적으로 작업하고 앱 테마 변경도 괜찮다고 생각되면 작업해보겠습니다!!

**Reply by YunSeok-Choi** [reply]
네! 저는 앨리가 올려주신 다크 테마도 되게 좋은 것 같아요! 
햔재 가능한 부분을 끝내고 추후에 다른 태스크도 이슈 생성하면 좋을 것 같아요! 
피그마도 확인하고 코멘트 달겠습니다!

**Comment by JIWON1923** [comment]
지금 작업하시는게 글을 이미지로 저장할 때 사용하는 테마 작업이 맞을까요 ?

**Reply by JMM00** [reply]
오....제가 애초에 회의때부터 앱 테마 선택이라고 착각하고 있었네요...!!!!! 이미지 배경 테마에서....테마 변경 아이콘은 문제가 되지 않는군요...고민 많이 했는데 전혀 다른 작업이었어요...ㅠㅠ 이왕 만든 김에 다크모드 적용에 대해서는 어떻게 생각하시나요..?

**Reply by JIWON1923** [reply]
다크모드 적용도 좋고, 테마 적용도 너무 좋아요! (만들어주신 색도 너어무 예쁩니다, 계절과 날씨도 아이디어 최곱니다)
다만 지금 당장 테마를 적용하기 위해서 설정을 어떤 화면에서 제공해야할지(아이콘은 뭐로 해야 좋을지 등..)는 같이 더 신중하게 논의해야할 것 같습니다
지민님이 고민해주신 내용들은 다음 스프린트에서 적용할 수도 있을 것 같네요! 최고 최고!

저는 테마, 다크모드 모두 적용할 수 있다고 생각해요! (편하신대로 작업해주세용)
라이트모드일 때에는 선택한 테마로, 다크모드일 때에는 다크모드를 제공할 수 있지 않을까요!? (착각이라면 지송)

그래서 제 결론을 정리해서 말씀드리자면!
- 우선 이번 스프린트에서는 이미지 저장할 때, 테마를 선택할 수 있는 기능을 먼저 제공한다.
- 다크모드 적용 찬성입니다!
- 추후 테마가 추가되더라도, 라이트모드일 때에만 테마를 적용하면 될 것 같습니다
- 테마 선택에 대한 아이콘 등은 같이 작업하고, 피드백 받으면서 다음 스프린트에서 진행했으면 합니다!!

**Reply by JMM00** [reply]
의견 감사합니다!! 우선 빠르게 이미지 저장 시, 테마 선택 기능을 먼저 작업 후 다시 한번 피드백 받고 진행하겠습니다~ 실시간 진행사항은 라라시>NewDesign>앱테마 만들어보기  참고해주세요~!
